src/include/function/list/operations/list_slice_operation.h

```diff
@@ -30,8 +30,12 @@ struct ListSlice {
 
     static inline void operation(ku_string_t& str, int64_t& begin, int64_t& end,
         ku_string_t& result, ValueVector& resultValueVector) {
-        int64_t startIdx = (begin == 0) ? 1 : begin;
-        int64_t endIdx = (end == 0) ? str.len : end;
+        int64_t startIdx = (begin == 0) ? 1 : ((begin < 0) ? (begin + str.len + 1) : begin);
```

Comment:
the logic handling negative index seems to be same for list extract and slice. So refactor a list operation utils class/struct to hold this logic.